pkg/webui/console/components/notifications/templates/client-requested.js

```diff
@@ -0,0 +1,191 @@
+    return 'organization'
+  }
+
+  return 'user'
+}
+
+const getId = entity => {
+  if ('organization_ids' in entity) {
+    return entity.organization_ids.organization_id
+  } else if ('user_ids' in entity) {
+    return entity.user_ids.user_id
+  }
+
+  return entity.user_id
+}
+
+const ClientRequestedPreview = ({ notificationData }) => {
+  const { data, sender_ids } = notificationData
+  const client = 'create_client_request' in data ? data.create_client_request.client : data.client
+  const collaborator =
+    'create_client_request' in data ? data.create_client_request.collaborator : data.collaborator
+
+  return (
+    <Message
+      content={m.preview}
+      values={{
+        senderType: capitalizeMessage(getType(sender_ids)),
+        senderTypeMiddle: getType(sender_ids),
+        id: getId(sender_ids),
+        collaboratorType: getType(collaborator),
+        collaboratorId: getId(collaborator),
+        lineBreak: <br />,
+        clientId: client.ids.client_id,
+      }}
+    />
+  )
+}
+
+ClientRequestedPreview.propTypes = {
+  notificationData: PropTypes.shape({
+    data: PropTypes.shape({
+      create_client_request: PropTypes.shape({
+        client: PropTypes.shape({}),
+        collaborator: PropTypes.shape({}),
+      }),
+      client: PropTypes.shape({}),
+      collaborator: PropTypes.shape({}),
+    }).isRequired,
+    sender_ids: PropTypes.shape({}).isRequired,
+  }).isRequired,
+}
+
+const ClientRequestedTitle = () => <Message content={m.title} />
+
+const ClientRequested = ({ receiver, notificationData }) => {
+  const { data, sender_ids } = notificationData
+  const client = 'create_client_request' in data ? data.create_client_request.client : data.client
+  const collaborator =
+    'create_client_request' in data ? data.create_client_request.collaborator : data.collaborator
+
+  return (
+    <>
+      <Message content={m.greeting} values={{ receiverName: receiver }} component="p" />
+      <Message
+        content={m.body}
+        values={{
+          senderType: capitalizeMessage(getType(sender_ids)),
+          senderTypeMiddle: getType(sender_ids),
+          id: getId(sender_ids),
+          collaboratorType: getType(collaborator),
+          collaboratorId: getId(collaborator),
+          code: msg => <code>{msg}</code>,
+          b: msg => <b>{msg}</b>,
+          lineBreak: <br />,
+        }}
+        component="p"
+      />
+      <Message
+        component="p"
+        content={m.clientId}
+        values={{
+          b: msg => <b>{msg}</b>,
+          code: msg => <code>{msg}</code>,
+          clientId: client.ids.client_id,
+        }}
+      />
+      <Message
+        content={m.link}
+        values={{
+          Link: msg => (
+            <Link.Anchor href={`${accountUrl}/oauth-clients/${client.ids.client_id}`} external>
+              {msg}
+            </Link.Anchor>
+          ),
+        }}
+      />
+    </>
+  )
+}
+
+ClientRequested.propTypes = {
```

Comment:
There's a lot of duplication here as well which can be outsourced to the prop-types module.